src/main/java/use_cases/UserManager.java

```diff
@@ -1,4 +1,62 @@
 package use_cases;
 
+import java.util.HashMap;
+import java.util.ArrayList;
+
+import entities.User;
+import entities.Post;
+
 public class UserManager {
+
+    /**
+     * Update User's username into username given
+     */
+    public void update_username(User user, String username) {
+        user.username = username;
+    }
+
+    /**
+     * Update User's password into password given
+     */
+    public void update_password(User user, String new_password) {
+        user.password = new_password;
+    }
+
+    /**
+     * Update User's bio into bio given
+     */
+    public void update_bio(User user, String bio) {
+        user.bio = bio;
+    }
+
+    /**
+     * Add 1 like to the cuisine category that User liked
+     */
+    public void add_like(User user, String cuisine_category) {
+        int current_likes = user.like_history.get(cuisine_category);
+        current_likes = current_likes + 1;
+        user.like_history.put(cuisine_category, current_likes);
+    }
+
+    /**
+     * follow User want_to_follow, adding want_to_follow into the logged-in user's following list
+     */
+    public String follow_user(User current_user, User want_to_follow) {
```

Comment:
It might be cleaner to return a Boolean instead of a String (true when we follow, false when we don't follow)

Reply:
agreed